app/code/Magento/InventoryCatalog/Model/GetProductIdsBySkus.php

```diff
@@ -21,20 +23,42 @@ class GetProductIdsBySkus implements GetProductIdsBySkusInterface
      */
     private $productResource;
 
+    /**
+     * @var CacheInterface
+     */
+    private $cache;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
     /**
      * @param ProductResourceModel $productResource
+     * @param CacheInterface $cache
+     * @param SerializerInterface $serializer
      */
     public function __construct(
-        ProductResourceModel $productResource
+        ProductResourceModel $productResource,
+        CacheInterface $cache,
+        SerializerInterface $serializer
```

Comment:
all dependencies to released classes should be introduced as optional to follow backward compatibility requirements